frontend/src/components/SettingsAccount.vue

```diff
@@ -66,34 +67,60 @@ const refreshData = async () => Promise.all([
   getAvailableEmails(),
 ]);
 
-// save user data
-const errorUsername = ref(false);
+// Form validation
+const errorUsername = ref<string>(null);
+const errorDisplayName = ref<string>(null);
+
+// Save user data
 const updateUser = async () => {
   const inputData = {
     username: activeUsername.value,
     name: activeDisplayName.value,
     secondary_email: activePreferredEmail.value,
   };
+
   const { data, error }: SubscriberResponse = await call('me').put(inputData).json();
   if (!error.value) {
     // update user in store
     user.updateProfile(data.value);
     await user.updateSignedUrl(call);
-    errorUsername.value = false;
+    errorUsername.value = null;
+    errorDisplayName.value = null;
     // TODO show some confirmation
     await refreshData();
   } else {
-    errorUsername.value = true;
+    errorUsername.value = t('error.usernameIsNotAvailable');
   }
 
   closeModals();
 };
 
 /**
+ * Validate user input first and only open modal if no errors occured.
  * Check if the username has been changed, and open a modal to warn the user their short link is going to change
  * If it didn't change, then just update the user immediately.
  */
 const updateUserCheckForConfirmation = async () => {
+  errorUsername.value = null;
+  errorDisplayName.value = null;
+
+  // Validate username
+  if (activeUsername.value === '') {
+    errorUsername.value = t('validation.fieldIsRequired', { field: t('label.username') });
```

Comment:
I feel like this should already trigger with the required attribute, since we use html form validation.

Reply:
At this place, we don't 😅 We just open the modal for confirmation of the username change...

Reply:
@MelissaAutumn I can change that to use form validation next week if you like!

Reply:
Sounds good to me, thank you!

Reply:
I added form validation at this place now.